docs/en/latest/plugins/ai-content-moderation.md

```diff
@@ -0,0 +1,220 @@
+| provider.aws_comprehend.region            | Yes          | String   | AWS region                                                                                                                               |
+| provider.aws_comprehend.endpoint          | No           | String   | AWS Comprehend service endpoint. Must match the pattern `^https?://`                                                                     |
+| moderation_categories                     | No           | Object   | Configuration for moderation categories. Must be one of: PROFANITY, HATE_SPEECH, INSULT, HARASSMENT_OR_ABUSE, SEXUAL, VIOLENCE_OR_THREAT |
+| toxicity_level                            | No           | Number   | Threshold for overall toxicity detection. Range: 0 - 1. Default: 0.5                                                                     |
+| llm_provider                              | Yes          | String   | Name of the LLM provider that this route will proxy requests to.                                                                         |
+
+## Example usage
+
+Create a route with the `ai-content-moderation` and `ai-proxy` plugin like so:
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes/1" -X PUT \
+  -H "X-API-KEY: ${ADMIN_API_KEY}" \
+  -d '{
+    "uri": "/post",
+    "plugins": {
+      "ai-content-moderation": {
+        "provider": {
+          "aws_comprehend": {
+            "access_key_id": "some_access_key",
+            "secret_access_key": "some_secret_key",
+            "region": "us-east-1"
+          }
+        },
+        "moderation_categories": {
+          "PROFANITY": 0.5
+        },
+        "llm_provider": "openai"
+      },
+      "ai-proxy": {
+        "auth": {
+          "header": {
+            "Authorization": "Bearer token"
+          }
+        },
+        "model": {
+          "provider": "openai",
+          "name": "gpt-4",
+          "options": {
+            "max_tokens": 512,
+            "temperature": 1.0
+          }
+        }
+      }
+    },
+    "upstream": {
+      "type": "roundrobin",
+      "nodes": {
+        "httpbin.org:80": 1
+      }
+    }
+  }'
+```
+
+The `ai-proxy` plugin is used here as it simplifies access to LLMs. However, you may configure the LLM in the upstream configuration as well.
+
+Now send a request:
+
+```shell
+curl http://127.0.0.1:9080/post -i -XPOST  -H 'Content-Type: application/json' -d '{
+  "info": "<some very seriously profane message>"
+}'
+```
+
+Then the request will be blocked with error like this:
+
+```text
+HTTP/1.1 400 Bad Request
+Date: Fri, 30 Aug 2024 11:21:21 GMT
+Content-Type: text/plain; charset=utf-8
+Transfer-Encoding: chunked
+Connection: keep-alive
+Server: APISIX/3.10.0
+
+request body exceeds toxicity threshold
+```
+
+Send a request with normal request body:
+
+```shell
+curl http://127.0.0.1:9080/post -i -XPOST  -H 'Content-Type: application/json' -d 'APISIX is wonderful'
```

Comment:
The opening paragraph says `This plugin must be used in routes that proxy requests to LLMs only` yet the example does not involve proxying to LLM. It feels a bit self-conflicting.

The example demonstrates exactly the integration could be used for checking requests NOT proxying to LLM.